### PR TITLE
Check for Gemfile in Ruby apps

### PR DESCRIPTION
**Pull request by sshaw**
RE #91. WIP. 

The code in `nxt_ruby.c` seems to prefer using Ruby C API calls over libc calls and calling methods via `rb_const_get` + `rb_funcall` instead of calling their C equivalents so this is what I've done. 

Additional code is needed as currently [this does not get executed](https://github.com/bundler/bundler/blob/257fb54da0003c3a67f6e7b3b5a242a4ba9c45cb/lib/bundler/setup.rb#L5). 

Possibly two options to fix: 

1. `chdir` do `script`'s root directory
1. Set the `BUNDLE_GEMFILE` environment variable 

I can investigate more later today. 

I will add tests once we agree on approach and coding style 😄

### CODE REVIEW

File: src/ruby/nxt_ruby.c [hunk 1]
```diff
@@ -191,6 +192,24 @@ nxt_ruby_init_io(nxt_task_t *task)
...
+nxt_ruby_bundler_project(nxt_str_t *script)
+{
+  VALUE file, dirname, path;
+
+  path = rb_str_new((const char *) script->start, (long) script->length);
+
+  file = rb_const_get(rb_cObject, rb_intern("File"));
+  dirname = rb_funcall(file, rb_intern("dirname"), 1, path);
+
+  rb_str_free(path);
```
Comment (sshaw) [inline src/ruby/nxt_ruby.c]: Is this necessary?

**Comment by sshaw** [comment]
> In config we can set working_directory (chdir), some like a:

The server should be able to figure this out. For a user I think this is a bit annoying having to set this. One would only have to set this if `script` lived outside the working directory. I don't think this is a valid use case when it comes to Rack apps. 

It makes sense to add this to `nxt_ruby_rack_init_t`.

> I added little changes. Try this path:

I will apply the patch tomorrow but key thing is still missing: 

 `chdir` to `working_dir` or set the BUNDLE_GEMFILE. I don't know enough about unit to know what problems a `chdir` may cause, so I guess that leaves `BUNDLE_GEMFILE`.

**Comment by sshaw** [comment]
And patch seg faults.

**Comment by lexborisov** [comment]
I forgot to make `qrefresh` before export patch. I updated the patch.

We use `working_directory` for all modules. Actually, [`working_directory`](https://github.com/nginx/unit/blob/2f3b9d458328c7c96f91259c1892810c16cbccdd/src/nxt_application.c#L330) is the same as `chdir`.
I'll think about using the `BUNDLE_GEMFILE` environment variable.

**Comment by romanm-odysii** [comment]
Is anything that prevents from this PR merged into master?

**Comment by VBart** [comment]
@romanm-odysii Please, see the above discussion. There's no consensus about how to configure a bundler gemfile and the patch isn't complete.

**Pull request closed without merging**